app/directives/proposal/proposal.directive.js

```diff
@@ -0,0 +1,140 @@
+            $log.debug("dbProposal: CONTROLLER");
+            var self = this;
+
+            self.testUserGuid = "926eee0b-8e94-4a14-beec-d9b60590547f"; // Elina Aalto
+
+            self.mode = PROP.MODE.COLLAPSED;
+            self.mds = PROP.MODE;
+            self.isPublished = false;
+            self.isUser = false;
+            self.editedText = "";
+            self.smallEditor = {
+                'menu': [
+                    ['bold', 'italic', 'underline', 'strikethrough', 'subscript', 'superscript']
+                ]
+            };
+            self.editor = {
+                'menu': [
+                    // ['bold', 'italic', 'underline', 'strikethrough', 'subscript', 'superscript'],
+                    // ['format-block'],
+                    // ['font'],
+                    // ['font-size'],
+                    // ['font-color', 'hilite-color'],
+                    // ['remove-format'],
+                    // ['ordered-list', 'unordered-list', 'outdent', 'indent'],
+                    // ['left-justify', 'center-justify', 'right-justify']
+                ]
+            };
+
+            function setProposal(proposal) {
+                if (proposal instanceof Object) {
+                    self.isPublished = (proposal.isPublished === PROPS.PUBLISHED.YES);
+                    self.isUser = (proposal.personGuid === self.testUserGuid);
+                    self.editedText = proposal.text;
+
+                    if (proposal.isNew) {
+                        self.mode = PROP.MODE.EDITOR;
+                        delete proposal.isNew;
+                    }
+                }
+            }
+
+            self.startEditing = function() {
+                self.editedText = $scope.proposal.text;
+                self.mode = PROP.MODE.EDITOR;
+            };
+
+            self.endEditing = function() {
+                if ($scope.proposal.text !== self.editedText) {
+                    $scope.proposal.text = self.editedText;
+                    $scope.onPost({ proposal: $scope.proposal });
+                }
+                self.mode = PROP.MODE.OPEN;
+            };
+
+            self.cancelEditing = function() {
+                self.mode = (self.editedText && self.editedText.lenght) ? PROP.MODE.OPEN : PROP.MODE.COLLAPSED;
+            };
+
+            self.deleteProposal = function() {
+                $scope.onDelete({ proposal: $scope.proposal });
+            };
+
+            self.shareProposal = function() {
+                $scope.proposal.isPublished = PROPS.PUBLISHED.YES;
+                $scope.onPost({ proposal: $scope.proposal });
+            };
+
+            self.open = function() {
+                self.mode = PROP.MODE.OPEN;
+            };
+
+            self.collapse = function() {
+                self.mode = PROP.MODE.COLLAPSED;
+            };
+
+            self.typeText = function(value) {
+                for (var index = 0; index < PROPS.TYPE.length; index++) {
```

Comment:
Here one could consider using the rootScope util function `objWithVal` which is already used by
e.g. opensignreqs.directive.js and signitem.ctr.js